### PR TITLE
fix(pkg/client): Prevent updates with incorrect database settings

### PR DESCRIPTION
**Pull request by byo**
Signed-off-by: Bartłomiej Święcki <bart@codenotary.com>

### CODE REVIEW

File: embedded/store/immustore.go [hunk 1]
```diff
@@ -173,8 +174,9 @@ type refVLog struct {
 }
 
 func Open(path string, opts *Options) (*ImmuStore, error) {
-	if !validOptions(opts) {
-		return nil, ErrIllegalArguments
+	err := opts.Validate()
```
Comment (jeroiraz) [inline embedded/store/immustore.go:177]: originally it was catching up the case when opts was nil, now it will panic...

Comment (byo) [reply embedded/store/immustore.go:177]: It will still catch this case, `Validate` is doing this check too.

**Review by jeroiraz — DISMISSED** [review]

**Comment by byo** [comment]
@jeroiraz I heavily modified server_tests.go file - after my changes it started showing issues with tests. I think I've found the reason of sporadic errors with `address already in use` errors here too. Please re-review.

File: embedded/store/options.go [hunk 2]
```diff
@@ -159,49 +160,125 @@ func DefaultIndexOptions() *IndexOptions {
...
+	}
+	if opts.WriteTxHeaderVersion > MaxTxHeaderVersion {
+		return fmt.Errorf("%w: invalid WriteTxHeaderVersion", ErrInvalidOptions)
+	}
+
+	// options below are only set during initialization and stored as metadata
+	if opts.MaxTxEntries <= 0 {
+		return fmt.Errorf("%w: invalid MaxTxEntries", ErrInvalidOptions)
+	}
+	if opts.MaxKeyLen <= 0 {
```
Comment (jeroiraz) [inline embedded/store/options.go:215]: min and max values may be unified in a single validation

File: embedded/store/options.go [hunk 3]
```diff
@@ -159,49 +160,125 @@ func DefaultIndexOptions() *IndexOptions {
...
 
-		opts.VLogMaxOpenedFiles > 0 &&
-		opts.TxLogMaxOpenedFiles > 0 &&
-		opts.CommitLogMaxOpenedFiles > 0 &&
+	if opts.MaxConcurrency <= 0 {
+		return fmt.Errorf("%w: invalid MaxConcurrency", ErrInvalidOptions)
+	}
 
-		opts.TxLogCacheSize >= 0 &&
+	if opts.MaxIOConcurrency <= 0 {
```
Comment (jeroiraz) [inline embedded/store/options.go:172]: min and max values may be unified in a single validation

File: embedded/store/options.go [hunk 4]
```diff
@@ -159,49 +160,125 @@ func DefaultIndexOptions() *IndexOptions {
...
-		opts.MaxNodeSize > 0 &&
-		opts.RenewSnapRootAfter >= 0 &&
-		opts.NodesLogMaxOpenedFiles > 0 &&
-		opts.HistoryLogMaxOpenedFiles > 0 &&
-		opts.CommitLogMaxOpenedFiles > 0
+	if opts.TimeFunc == nil {
+		return fmt.Errorf("%w: invalid TimeFunc", ErrInvalidOptions)
+	}
+
+	if opts.WriteTxHeaderVersion < 0 {
```
Comment (jeroiraz) [inline embedded/store/options.go:204]: min and max values may be unified in a single validation

**Pull request merged**